### PR TITLE
Feature/pim 1640 change forage dop summary table backend

### PR DESCRIPTION
**Pull request by mhostett**

### CODE REVIEW

File: cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		FROM declared_yield_contract dyc
+		JOIN grower_contract_year gcy ON gcy.contract_id = dyc.contract_id AND gcy.crop_year = dyc.crop_year
+		JOIN contracted_field_detail cfd ON cfd.grower_contract_year_id = gcy.grower_contract_year_id
+		JOIN annual_field_detail afd ON afd.annual_field_detail_id = cfd.annual_field_detail_id
+		JOIN inventory_field inv ON inv.field_id = afd.field_id AND inv.crop_year = afd.crop_year
+		JOIN inventory_seeded_forage isd ON isd.inventory_field_guid = inv.inventory_field_guid
+		JOIN declared_yield_field_forage dyff ON dyff.inventory_field_guid = inv.inventory_field_guid
+		WHERE dyc.declared_yield_contract_guid = dyccf.declared_yield_contract_guid
+		AND isd.commodity_type_code = dyccf.commodity_type_code
+		GROUP BY inv.crop_year, isd.commodity_type_code
```
Comment (vivid-esnauffer) [inline cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql:30]: What if the percentage moisture is different between the different cuts? Won't that affect the Quantity Harvested Acres calculation?

Comment (vivid-esnauffer) [inline cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql:30]: What if the percentage moisture is different for the different cuts? Won't that affect the way quantity_harvested_tons are calculated?

File: cirras-underwriting-api/cirras-underwriting-service-api/src/main/java/ca/bc/gov/mal/cirras/underwriting/service/api/v1/impl/CirrasDopYieldServiceImpl.java [hunk 2]
```diff
@@ -519,6 +534,9 @@ public DopYieldContract<? extends AnnualField> createDopYieldContract(
 
 				//Calculate and save commodity totals
 				updateDeclaredYieldContractCommodityForage(declaredYieldContractGuid, dopYieldContract, userId);
+				
+				//Calculate and save field rollups
+				updateDeclaredYieldRollupForage(declaredYieldContractGuid, dopYieldContract, userId);
```
Comment (vivid-esnauffer) [inline cirras-underwriting-api/cirras-underwriting-service-api/src/main/java/ca/bc/gov/mal/cirras/underwriting/service/api/v1/impl/CirrasDopYieldServiceImpl.java:540]: I don't see `updateDeclaredYieldRollupForage` executed in `updateDopYieldContract` method

File: cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql (hunk 1 above)
Comment (mhostett) [reply cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql:30]: We're not calculating quantity harvested tons and yield/acre. This script only copies over all records from declared_yield_contract_cmdty_forage to declared_yield_rollup_forage and calculate of the total bales for a commodity type.
The calculation of the rollup in the app is done in the backend and considers the moisture and weight.

Comment (vivid-esnauffer) [reply cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql:30]: So, for example for a 2024 policy, there won't be calculations for quantity harvested tons and yield/acre in the summary table until they try to save the DOP again?

**Review by vivid-esnauffer — APPROVED** [review]

File: cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql (hunk 1 above)
Comment (mhostett) [reply cirras-underwriting-liquibase/scripts/03_00_xx/03_05_00/00/dml/cuws.insert.declared_yield_rollup_forage.sql:30]: Yes, if they don't update the DOP there is no need to update the values or recalculate. The values are already calculated and copied over. Not sure I understand your question.

**Pull request merged**